odl/discr/diff_ops.py

```diff
@@ -774,26 +774,27 @@ def finite_diff(f, axis=0, dx=1.0, method='forward', out=None, **kwargs):
     ndim = f_arr.ndim
 
     if f_arr.shape[axis] < 2:
-        raise ValueError('In axis {}: at least two elements required, got {}.'
+        raise ValueError('in axis {}: at least two elements required, got {}'
                          ''.format(axis, f_arr.shape[axis]))
 
     if axis < 0:
         axis += ndim
     if not (0 <= axis < ndim):
-        raise IndexError('axis {} outside the valid range 0 ... {}'
+        raise IndexError('`axis` {} outside the valid range 0 ... {}'
```

Comment:
I write this here, but actually it's a separate issue.

This error message is problematic (I think I wrote it) since it contradicts the code and the doc which allows negative indices. We should adhere to a style similar to NumPy's:
```python
>>> np.ones(1)[2]
IndexError: index 2 is out of bounds for axis 0 with size 1
>>> np.ones(1)[-2]
IndexError: index -2 is out of bounds for axis 0 with size 1
>>> np.sum(np.ones(1), axis=2)
ValueError: 'axis' entry is out of bounds
```

We could simply say
```
`axis` {} is out of bounds
```